haystack/preprocessor/utils.py

```diff
@@ -86,32 +90,46 @@ def convert_files_to_dicts(dir_path: str, clean_func: Optional[Callable] = None,
     """
 
     file_paths = [p for p in Path(dir_path).glob("**/*")]
-    if ".pdf" in [p.suffix.lower() for p in file_paths]:
-        pdf_converter = PDFToTextConverter()  # type: Optional[PDFToTextConverter]
-    else:
-        pdf_converter = None
+    allowed_suffixes = [".pdf", ".txt", ".docx"]
+    converter_dict: Dict[str, BaseConverter] = {}
 
-    documents = []
+    file_ext_dict: Dict[str, List[Path]] = {}
     for path in file_paths:
-        if path.suffix.lower() == ".txt":
-            with open(path) as doc:
-                text = doc.read()
-        elif path.suffix.lower() == ".pdf" and pdf_converter:
-            document = pdf_converter.convert(path)
+        file_suffix = path.suffix.lower()
+        if file_suffix in allowed_suffixes:
+            if file_suffix not in file_ext_dict:
+                file_ext_dict[file_suffix] = []
+            file_ext_dict[file_suffix].append(path)
+        elif not path.is_dir():
+            logger.warning('Skipped file {0} as type {1} is not supported here. '
+                           'See haystack.file_converter for support of more file types'.format(path, file_suffix))
+
+    # No need to initialize converter if file type not present
+    for file_suffix in file_ext_dict.keys():
+        if file_suffix in ".pdf":
```

Comment:
Why `in` and not `==`? Might cause trouble with files like ".pd"

Reply:
Yes correct I missed this. Corrected now.